be/src/util/disk_info.cpp

```diff
@@ -43,29 +43,48 @@ int DiskInfo::_s_num_datanode_dirs;
 void DiskInfo::get_device_names() {
     // Format of this file is:
     //    major, minor, #blocks, name
-    // We are only interesting in name which is formatted as device_name<partition #>
-    // The same device will show up multiple times for each partition (e.g. sda1, sda2).
+    // We are only want to get unique device name from /proc/partitions.
+    // The same device may show up multiple times for each partition (e.g. sda1, sda2), in this case, we only use sda for device name.
+    // In another case like nvme0n1, this device only show up once in /proc/partitions, then we use nvme0n1 for device name.
+    std::map<std::string, int> disk_prefix_count;
     std::ifstream partitions("/proc/partitions", std::ios::in);
 
     while (partitions.good() && !partitions.eof()) {
         std::string line;
         getline(partitions, line);
         boost::trim(line);
-
         std::vector<std::string> fields = strings::Split(line, " ", strings::SkipWhitespace());
-
         if (fields.size() != 4) {
             continue;
         }
-
         std::string name = fields[3];
-
         if (name == "name") {
             continue;
         }
+        std::string prefix = name.substr(0, name.find_last_not_of("0123456789") + 1);
+        disk_prefix_count[prefix]++;
+    }
+
+    // Read /proc/partitions again to get more detail information.
+    partitions.clear();
```

Comment:
need to add more tests.
1. split this function into two. the other one will take all the contents as input and split into the device map
2. add more tests to cover the second function.
Otherwise, this will be buggy.